src/core_plugins/elasticsearch/lib/kibana_index_patches/patch_missing_title_keyword_fields.js

```diff
@@ -0,0 +1,68 @@
+import { getRootProperties } from '../../../../server/mappings';
+import { get } from 'lodash';
+import { KibanaIndexPatch } from './kibana_index_patch';
+
+const propertiesWithTitles = [
+  'index-pattern',
+  'dashboard',
+  'visualization',
+  'search',
+];
```

Comment:
Why not apply this all properties with a `title` field?

Reply:
Using [`getProperty`](https://github.com/elastic/kibana/blob/f839564d9190f52d48d8f04bbfac149fd1804c72/src/server/mappings/lib/get_property.js#L35) this could be something like:

```js
const properties = getRootProperties(mappings)
const propertiesWithTitles = Object.keys(properties)
  .filter(property => Boolean(getProperty(mappings, [property, 'title', 'keyword'])))
```

Reply:
Love it!